rust/src/logs/conversion.rs

```diff
@@ -0,0 +1,159 @@
+    fn from(value: &(opentelemetry::Key, opentelemetry::logs::AnyValue)) -> Self {
+        Self {
+            key: value.0.to_string(),
+            value: value.1.clone().into(),
+        }
+    }
+}
+
+// The Rust SDK currently keeps the log `AnyValue` type separate from the general purpose `KeyValue`->`Value` type.
+// Per the OTel spec, they will likely be the same at some point.
+impl From<&opentelemetry::logs::AnyValue> for AnyValue {
+    fn from(value: &opentelemetry::logs::AnyValue) -> Self {
+        use opentelemetry::logs as otel;
+        match value {
+            otel::AnyValue::Boolean(v) => AnyValue::Value(AnyType::Boolean(v.clone())),
+            otel::AnyValue::Double(v) => AnyValue::Value(AnyType::Double(v.clone())),
+            otel::AnyValue::Int(v) => AnyValue::Value(AnyType::Integer(v.clone())),
+            otel::AnyValue::String(v) => AnyValue::Value(AnyType::String(v.to_string())),
+            otel::AnyValue::Bytes(v) => AnyValue::Value(AnyType::Bytes(v.to_vec())),
+            otel::AnyValue::ListAny(v) => AnyValue::List(
+                v.iter()
+                    .map(|e| match e {
+                        otel::AnyValue::Boolean(v) => AnyType::Boolean(v.clone()),
+                        otel::AnyValue::Double(v) => AnyType::Double(v.clone()),
+                        otel::AnyValue::Int(v) => AnyType::Integer(v.clone()),
+                        otel::AnyValue::String(v) => AnyType::String(v.to_string()),
+                        otel::AnyValue::Bytes(v) => AnyType::Bytes(v.to_vec()),
+                        // The `ListAny` and `Map` types are intentionally omitted because WIT doesn't support infinite recursion.
+                        _ => unimplemented!(),
+                    })
+                    .collect(),
+            ),
+            otel::AnyValue::Map(map) => {
+                let mut result: Vec<(String, AnyType)> = vec![];
+                for (k, v) in map.iter() {
+                    result.push((
+                        k.to_string(),
+                        match v {
+                            otel::AnyValue::Boolean(v) => AnyType::Boolean(v.clone()),
+                            otel::AnyValue::Double(v) => AnyType::Double(v.clone()),
+                            otel::AnyValue::Int(v) => AnyType::Integer(v.clone()),
+                            otel::AnyValue::String(v) => AnyType::String(v.to_string()),
+                            otel::AnyValue::Bytes(v) => AnyType::Bytes(v.to_vec()),
+                            // The `ListAny` and `Map` types are intentionally omitted because WIT doesn't support infinite recursion.
+                            _ => unimplemented!(),
+                        },
+                    ));
+                }
+                AnyValue::Map(result)
+            }
+            _ => unimplemented!(),
+        }
+    }
+}
+
+impl From<opentelemetry::logs::AnyValue> for AnyValue {
```

Comment:
Why do we need this twice. Is it because one is a reference?

Could we work around that by just referencing whatever we're converting where it is an owned type?